index.js

```diff
@@ -1,48 +1,32 @@
 'use strict';
 
 var path = require('path');
+var bowerConfig = require('bower/lib/config');
 
-var isFileSync = require('is-file').sync;
-var readJson = require('read-json');
-var readJsonSync = require('read-json-sync');
+function bowerDirectorySync(option) {
+  option = option || {};
+
+  option.cwd = option.cwd || process.cwd();
 
-function getBowerDir(json) {
-  return path.join(json.cwd || '', json.directory || 'bower_components');
+  return path.resolve(option.cwd, bowerConfig(option).directory);
 }
 
-module.exports = function bowerDirectory(option, cb) {
+function bowerDirectory(option, cb) {
+  var p;
+  var err;
+
   if (cb === undefined) {
     cb = option;
   }
-  option = option || {};
-  var resolveFromCwd = path.resolve.bind(path, option.cwd || '');
-
-  readJson(resolveFromCwd('.bowerrc'), function(err, data) {
-    var bowerDirRelative;
-
-    if (!err) {
-      bowerDirRelative = getBowerDir(data);
-    } else {
-      if (err.name === 'SyntaxError') {
-        cb(err);
-        return;
-      }
-
-      bowerDirRelative = 'bower_components';
-    }
-    cb(null, resolveFromCwd(bowerDirRelative));
-  });
-};
-
-module.exports.sync = function bowerDirectorySync(option) {
-  option = option || {};
-  var resolveFromCwd = path.resolve.bind(path, option.cwd || '');
-
-  var bowerrcPath = resolveFromCwd('.bowerrc');
 
-  if (isFileSync(bowerrcPath)) {
-    return resolveFromCwd(getBowerDir(readJsonSync(bowerrcPath)));
-  } else {
-    return resolveFromCwd('bower_components');
+  try {
+    p = bowerDirectorySync(option);
+  } catch (e) {
+    err = e;
+  } finally {
+    cb(err, p);
```

Comment:
I want `bowerDirectory(option, cb)` to work as an asynchronous function, but this change makes it synchronous...